### PR TITLE
Use GKE specific configuration in startup scripts in GKE deployment.

### PR DESCRIPTION
**Pull request by benhxy**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Allow GKE to configure VM start up differently.
More specifically, kubeconfig files and kube-apiserver SNI cert configuration is changed for GKE.

#### Which issue(s) this PR fixes:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs

```

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -1269,6 +1269,10 @@ EOF
 function create-kubeconfig {
   local component=$1
   local token=$2
+
+  if [[ -e "${KUBE_HOME}/bin/gke-internal-configure-helper.sh" ]]; then
+    gke-internal-create-kubeconfig ${component} ${token}
+  else
```
Comment (PurelyApplied) [inline cluster/gce/gci/configure-helper.sh:1275]: Nit: please indent the now-`else` clause.

**Review by PurelyApplied — COMMENTED** [review]
I wonder if it works when the comment is embedded in a Review...

/ok-to-test

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (benhxy) [reply cluster/gce/gci/configure-helper.sh:1275]: Thanks. Fixed the indentation.

**Review by logicalhan — COMMENTED** [review]
The shellcheck failure seems a little too coincidental, would you mind running the shellcheck command locally?

**Comment by benhxy** [comment]
> The shellcheck failure seems a little too coincidental, would you mind running the shellcheck command locally?

You are spot on. Ran shellcheck and fixed the error.

**Review by logicalhan — COMMENTED** [review]
/lgtm

**Pull request closed without merging**